lib/src/http/managers/message_manager.dart

```diff
@@ -386,5 +390,88 @@ class MessageManager extends Manager<Message> {
     await client.httpHandler.executeSafe(request);
   }
 
+  /// Adds a reaction to a message.
+  Future<void> addReaction(Snowflake id, String emoji) async {
+    final route = HttpRoute()
+      ..channels(id: channelId.toString())
+      ..messages(id: id.toString())
+      ..reactions(emoji: _encodeForAPI(emoji), userId: '@me');
+
+    final request = BasicRequest(route, method: 'PUT');
+
+    await client.httpHandler.executeSafe(request);
+  }
+
+  /// Deletes our own reaction from a message.
+  Future<void> deleteOwnReaction(Snowflake id, String emoji) async {
+    final route = HttpRoute()
+      ..channels(id: channelId.toString())
+      ..messages(id: id.toString())
+      ..reactions(emoji: _encodeForAPI(emoji), userId: '@me');
+
+    final request = BasicRequest(route, method: 'DELETE');
+
+    await client.httpHandler.executeSafe(request);
+  }
+
+  /// Deletes all reactions on a message.
+  Future<void> deleteAllReactions(Snowflake id) async {
+    final route = HttpRoute()
+      ..channels(id: channelId.toString())
+      ..messages(id: id.toString())
+      ..reactions();
+    final request = BasicRequest(route, method: 'DELETE');
+
+    await client.httpHandler.executeSafe(request);
+  }
+
+  /// Deletes all reactions for a given emoji on a message.
+  Future<void> deleteReactionForUser(Snowflake id, Snowflake userId, String emoji) async {
+    final route = HttpRoute()
+      ..channels(id: channelId.toString())
+      ..messages(id: id.toString())
+      ..reactions(emoji: _encodeForAPI(emoji), userId: userId.toString());
+
+    final request = BasicRequest(route, method: 'DELETE');
+
+    await client.httpHandler.executeSafe(request);
+  }
+
+  /// Deletes all reations for a given emoji on a message.
+  Future<void> deleteReaction(Snowflake id, String emoji) async {
+    final route = HttpRoute()
+      ..channels(id: channelId.toString())
+      ..messages(id: id.toString())
+      ..reactions(emoji: _encodeForAPI(emoji));
+
+    final request = BasicRequest(route, method: 'DELETE');
+
+    await client.httpHandler.executeSafe(request);
+  }
+
+  /// Get a list of users that reacted with a given emoji on a message.
+  Future<List<User>> fetchReactions(Snowflake id, String emoji) async {
+    final route = HttpRoute()
+      ..channels(id: channelId.toString())
+      ..messages(id: id.toString())
+      ..reactions(emoji: _encodeForAPI(emoji));
+    final request = BasicRequest(route);
+
+    final response = await client.httpHandler.executeSafe(request);
+
+    return parseMany(response.jsonBody as List, client.users.parse);
+  }
```

Comment:
These methods should take a `ReactionBuilder` instead of a String that is then encoded. Makes it much safer,